### PR TITLE
239 validate interval parts

### PR DESCRIPTION
**Pull request by RakeshBobba03**
This pull request implements two new operators split_parts_have_equal_length and split_parts_have_not_equal_length to address issue https://github.com/cdisc-org/cdisc-rules-engine/issues/239, which validates that interval date/time representations have the same completeness on both sides of the solidus. The operators split a value by a configurable separator (defaulting to "/") and verify that both resulting parts have equal string length, correctly validating examples like "2003-12-15T10:00/2003-12-15T10:30" while flagging violations like "2003-12-15T10:00/2003-12-15T10:30:15" where precision differs. The implementation is intentionally generic to support any paired string format requiring equal precision, handles all edge cases including nulls and empty strings, and includes unit tests for both PandasDataset and DaskDataset along with updated operator documentation.

### CODE REVIEW

File: tests/unit/test_check_operators/test_string_comparison.py [hunk 1]
```diff
@@ -703,3 +703,133 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
...
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
+        # PandasDataset tests with "/" separator
+        # Combines SENDIG examples, invalid cases, edge cases, and multiple separators
+        (
+            {
+                "target": [
```
Comment (gerrycampion) [inline tests/unit/test_check_operators/test_string_comparison.py]: instead of defining a dictionary here like {"target": [1,2,3]}, you could just define the inner array like [1,2,3] and build the target below like:
df = dataset_type.from_dict({"target": data})
It just simplifies the test cases a bit. 

Same for other locations

File: tests/unit/test_check_operators/test_string_comparison.py [hunk 2]
```diff
@@ -703,3 +703,133 @@ def test_not_matches_regex(data, comparator, dataset_type, expected_result):
         {"target": "target", "comparator": comparator}
     )
     assert result.equals(df.convert_to_series(expected_result))
+
+
+@pytest.mark.parametrize(
+    "data,separator,dataset_type,expected_result",
+    [
+        # PandasDataset tests with "/" separator
+        # Combines SENDIG examples, invalid cases, edge cases, and multiple separators
```
Comment (gerrycampion) [inline tests/unit/test_check_operators/test_string_comparison.py]: The commenting is inconsistent and a lot of the comments aren't necessary when things are self-evident by the code and it may add noise and make the code harder to maintain. Leave comments for more complex code or high-level descriptions.

**Comment by RamilCDISC** [comment]
I have validated the new operator using dev editor. All seems fine. Please address @gerrycampion change requests and after I will add the closing comment of validation and merge.

**Review by gerrycampion — APPROVED** [review]
